### DISCUSSION TITLE
Integration of Allure reports in Katalon 9

### DISCUSSION THREAD

**Discussion by ArthurRallon** [allure-framework/allure2#2767] [Questions & Support]
Hello Allure community, 
I'm part of a QA team that uses Katalon for its automated BDD tests. To modernize our reports, we decided to turn to Allure, and I'm in charge of its implementation. 
Today, from within Katalon, I'm able to launch the Allure report generation command using the appropriate command. 

For the moment I'm limiting myself to 1 cucumber.json file to generate the report, but Allure can't extract the data from the file and returns an empty report.

I've made some enquiries and I'm under the impression that the cucumber.json file isn't in the right format, which would explain why Allure can't retrieve any information.

What do you think? Thanks in advance!

![image](https://github.com/user-attachments/assets/463e0929-1816-461f-afdb-d8903f66bb39)

**Comment by baev** [comment]
Allure has yet to support the `cucumber.json`. We plan to provide support, but the issue is low priority since we have proper integrations for the most used cucumber implementations (Ruby, Java, Javascript, PHP, Python, and C#).

Allure does not officially support Katalon products since Katalon is a closed source and requires a license to function properly (even the CI integration, which is necessary for open-source development, requires a license). 

However, since Katalon uses Cucumber Java under the hood, it should be possible to integrate it with Allure. 

You can follow the steps specified in the documentation: https://allurereport.org/docs/cucumberjvm/

1. Add the necessary dependencies. 
2. Enable the Allure Cucumber plugin
3. AspectJ configuration can be skipped since it only adds support for the `@Step` and `@Attachment` annotations, which can be replaced with `Allure.step` and `Allure.attachment` Runtime API calls.

To enable the Allure Cucumber plugin, you probably need to use a custom cucumber runner as described here: https://docs.katalon.com/katalon-studio/keywords/keyword-description-in-katalon-studio/cucumber-keywords/cucumber-run-with-cucumber-runner.

**Comment by montbaga** [comment]
You can now use the below tool to integrate your katalon project with allure . 

https://github.com/montbaga/allure-katalon-bridge
